Index: .idea/inspectionProfiles/Project_Default.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/inspectionProfiles/Project_Default.xml b/.idea/inspectionProfiles/Project_Default.xml
new file mode 100644
--- /dev/null	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
+++ b/.idea/inspectionProfiles/Project_Default.xml	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -0,0 +1,8 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="AutoCloseableResource" enabled="true" level="WARNING" enabled_by_default="true">
+      <option name="METHOD_MATCHER_CONFIG" value="java.util.Formatter,format,java.io.Writer,append,com.google.common.base.Preconditions,checkNotNull,org.hibernate.Session,close,java.io.PrintWriter,printf,java.io.PrintStream,printf,java.net.http.HttpClient,newHttpClient" />
+    </inspection_tool>
+  </profile>
+</component>
\ No newline at end of file
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision 71c77b861952d0759688968310484b6ec7429a5b)
+++ b/pom.xml	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -102,7 +102,11 @@
       <artifactId>junit-jupiter-engine</artifactId>
       <version>${junit.version}</version>
       <scope>test</scope>
-    </dependency>  </dependencies>
+    </dependency>
+
+  </dependencies>
+
+
 
   <build>
     <plugins>
Index: src/main/java/com/shavneva/billingsystemdesktop/ButtonController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/shavneva/billingsystemdesktop/ButtonController.java b/src/main/java/com/shavneva/billingsystemdesktop/ButtonController.java
--- a/src/main/java/com/shavneva/billingsystemdesktop/ButtonController.java	(revision 71c77b861952d0759688968310484b6ec7429a5b)
+++ b/src/main/java/com/shavneva/billingsystemdesktop/ButtonController.java	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -5,17 +5,40 @@
 import javafx.fxml.FXMLLoader;
 import javafx.scene.Node;
 import javafx.scene.Scene;
+import javafx.scene.control.PasswordField;
+import javafx.scene.control.TextField;
 import javafx.stage.Stage;
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.*;
+
+
 
 public class ButtonController {
+    @FXML
+    private TextField login;
+    @FXML
+    private TextField name;
+    @FXML
+    private TextField surname;
+    @FXML
+    private TextField phoneNumber;
+    @FXML
+    private TextField email;
+    @FXML
+    private PasswordField passwordField;
+    @FXML
+    private TextField repeatedPassword;
 
     @FXML
     protected void openRegistrationWindowClick(ActionEvent event) {
         try {
+
             Stage currentStage = (Stage) ((Node) event.getSource()).getScene().getWindow();
             currentStage.hide();
+
             FXMLLoader fxmlLoader = new FXMLLoader();
             fxmlLoader.setLocation(getClass().getResource("registration-view.fxml"));
 
@@ -26,27 +49,100 @@
             newStage.setTitle("Окно регистрации");
             newStage.setScene(scene);
             newStage.show();
-            }catch (IOException e){
+
+            } catch(IOException e){
+            ErrorDialog.showError("Произошла ошибка: " + e.getMessage());
+        }
+    }
+    @FXML
+    protected void openUWindowClick(ActionEvent event){
+        try {
+            String firstName = name.getText();
+            String lastName = surname.getText();
+            String registrationEmail = email.getText();
+            String pNumber =  phoneNumber.getText();
+            String password = passwordField.getText();
+            String rPassword = repeatedPassword.getText();
+
+            String requestBody = "{\"firstName\""+":"+"\""+firstName+"\""+","+
+                    "\"lastName\""+":"+"\""+lastName+"\""+","+
+                    "\"email\""+":"+"\""+registrationEmail+"\""+","+
+                    "\"number\""+":"+"\""+pNumber+"\""+","+
+                    "\"password\""+":"+"\""+password+"\""+
+                    "}";
+
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI("http://localhost:8080/users/createUser"))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                    .build();
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            System.out.println(response.statusCode());
+            System.out.println(response.body());
+
+            if(response.statusCode() == 200){
+                Stage currentStage = (Stage) ((Node) event.getSource()).getScene().getWindow();
+                currentStage.hide();
+
+                FXMLLoader fxmlLoader = new FXMLLoader();
+                fxmlLoader.setLocation(getClass().getResource("userwindow-view.fxml"));
+
+                Scene scene = new Scene(fxmlLoader.load(), 400, 290);
+                Stage newStage = new Stage();
+
+                newStage.setTitle("Биллинговая система");
+                newStage.setScene(scene);
+                newStage.show();
+            }
+        } catch (IOException e){
             ErrorDialog.showError("Произошла ошибка: " + e.getMessage());
+        } catch (URISyntaxException | InterruptedException e) {
+            throw new RuntimeException(e);
         }
     }
+
     @FXML
     protected void openUserWindowClick(ActionEvent event) {
-        try {
-            Stage currentStage = (Stage) ((Node) event.getSource()).getScene().getWindow();
-            currentStage.hide();
+        try{
+            String userName = login.getText();
+            String password = passwordField.getText();
+
+            String requestBody = "{\"login\""+":"+"\""+userName+"\""+","+"\"password\""+":"+"\""+password+"\""+
+                    "}";
+
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI("http://localhost:8080/users/login"))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                    .build();
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            System.out.println(response.statusCode());
+            System.out.println(response.body());
+
+            if(response.statusCode() == 200){
+                Stage currentStage = (Stage) ((Node) event.getSource()).getScene().getWindow();
+                currentStage.hide();
 
-            FXMLLoader fxmlLoader = new FXMLLoader();
-            fxmlLoader.setLocation(getClass().getResource("userwindow-view.fxml"));
+                FXMLLoader fxmlLoader = new FXMLLoader();
+                fxmlLoader.setLocation(getClass().getResource("userwindow-view.fxml"));
 
-            Scene scene = new Scene(fxmlLoader.load(), 400, 290);
-            Stage newStage = new Stage();
+                Scene scene = new Scene(fxmlLoader.load(), 400, 290);
+                Stage newStage = new Stage();
 
-            newStage.setTitle("Биллинговая система");
-            newStage.setScene(scene);
-            newStage.show();
-        }catch (IOException e){
+                newStage.setTitle("Биллинговая система");
+                newStage.setScene(scene);
+                newStage.show();
+            }
+           /* else{
+                JSONObject json = new JSONObject(response.body());
+                ErrorDialog.showError("Произошла ошибка: " + json.get);
+            }*/
+        } catch (IOException e){
             ErrorDialog.showError("Произошла ошибка: " + e.getMessage());
+        } catch (URISyntaxException | InterruptedException e) {
+            throw new RuntimeException(e);
         }
     }
 }
\ No newline at end of file
Index: src/main/java/com/shavneva/billingsystemdesktop/LoginApplication.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/shavneva/billingsystemdesktop/LoginApplication.java b/src/main/java/com/shavneva/billingsystemdesktop/LoginApplication.java
--- a/src/main/java/com/shavneva/billingsystemdesktop/LoginApplication.java	(revision 71c77b861952d0759688968310484b6ec7429a5b)
+++ b/src/main/java/com/shavneva/billingsystemdesktop/LoginApplication.java	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -5,9 +5,12 @@
 import javafx.scene.Scene;
 import javafx.stage.Stage;
 
+
+
 import java.io.IOException;
 
 public class LoginApplication extends Application {
+
     @Override
     public void start(Stage stage) throws IOException {
         FXMLLoader fxmlLoader = new FXMLLoader(LoginApplication.class.getResource("login-view.fxml"));
@@ -15,7 +18,6 @@
         stage.setTitle("Окно логинации");
         stage.setScene(scene);
         stage.show();
-
     }
 
     public static void main(String[] args) {
Index: src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/module-info.java b/src/main/java/module-info.java
--- a/src/main/java/module-info.java	(revision 71c77b861952d0759688968310484b6ec7429a5b)
+++ b/src/main/java/module-info.java	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -10,6 +10,8 @@
     requires org.kordamp.bootstrapfx.core;
     requires eu.hansolo.tilesfx;
     requires com.almasb.fxgl.all;
+    requires java.net.http;
+    requires com.fasterxml.jackson.databind;
 
     opens com.shavneva.billingsystemdesktop to javafx.fxml;
     exports com.shavneva.billingsystemdesktop;
Index: src/main/resources/com/shavneva/billingsystemdesktop/login-view.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/com/shavneva/billingsystemdesktop/login-view.fxml b/src/main/resources/com/shavneva/billingsystemdesktop/login-view.fxml
--- a/src/main/resources/com/shavneva/billingsystemdesktop/login-view.fxml	(revision 71c77b861952d0759688968310484b6ec7429a5b)
+++ b/src/main/resources/com/shavneva/billingsystemdesktop/login-view.fxml	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -25,8 +25,8 @@
               </rowConstraints>
                <children>
                   <Label prefHeight="32.0" prefWidth="234.0" text="Номер телефона или e-mail" />
-                  <TextField prefHeight="31.0" prefWidth="218.0" GridPane.columnIndex="1" />
-                  <PasswordField prefHeight="34.0" prefWidth="218.0" GridPane.columnIndex="1" GridPane.rowIndex="1" />
+                  <TextField fx:id="login" id="login" prefHeight="31.0" prefWidth="218.0" GridPane.columnIndex="1" />
+                  <PasswordField fx:id="passwordField" prefHeight="34.0" prefWidth="218.0" GridPane.columnIndex="1" GridPane.rowIndex="1" />
                   <Label text="Пароль" GridPane.rowIndex="1" />
                </children>
             </GridPane>
Index: src/main/resources/com/shavneva/billingsystemdesktop/registration-view.fxml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/com/shavneva/billingsystemdesktop/registration-view.fxml b/src/main/resources/com/shavneva/billingsystemdesktop/registration-view.fxml
--- a/src/main/resources/com/shavneva/billingsystemdesktop/registration-view.fxml	(revision 71c77b861952d0759688968310484b6ec7429a5b)
+++ b/src/main/resources/com/shavneva/billingsystemdesktop/registration-view.fxml	(revision fb39630bc0a747d95f867bad6f1ee2b97b1c7127)
@@ -30,14 +30,14 @@
             <Label text="Номер телефона" GridPane.rowIndex="2" />
             <Label text="Почта" GridPane.rowIndex="3" />
             <Label contentDisplay="CENTER" text="Пароль" GridPane.rowIndex="4" />
-            <TextField prefHeight="32.0" prefWidth="253.0" GridPane.columnIndex="1" />
-            <TextField GridPane.columnIndex="1" GridPane.rowIndex="1" />
-            <TextField GridPane.columnIndex="1" GridPane.rowIndex="2" />
-            <TextField GridPane.columnIndex="1" GridPane.rowIndex="3" />
-            <PasswordField GridPane.columnIndex="1" GridPane.rowIndex="4" />
-            <PasswordField GridPane.columnIndex="1" GridPane.rowIndex="5" />
+            <TextField fx:id="name" prefHeight="32.0" prefWidth="253.0" GridPane.columnIndex="1" />
+            <TextField fx:id="surname" GridPane.columnIndex="1" GridPane.rowIndex="1" />
+            <TextField fx:id="phoneNumber" GridPane.columnIndex="1" GridPane.rowIndex="2" />
+            <TextField fx:id="email" GridPane.columnIndex="1" GridPane.rowIndex="3" />
+            <PasswordField fx:id="passwordField" GridPane.columnIndex="1" GridPane.rowIndex="4" />
+            <PasswordField fx:id="repeatedPassword" GridPane.columnIndex="1" GridPane.rowIndex="5" />
          </children>
       </GridPane>
-      <Button mnemonicParsing="false" onAction="#openUserWindowClick" prefHeight="40.0" prefWidth="190.0" text="Зарегестрироваться" />
+      <Button mnemonicParsing="false" onAction="#openUWindowClick" prefHeight="40.0" prefWidth="190.0" text="Зарегестрироваться" />
    </children>
 </VBox>
